great_expectations/validator/validator.py

```diff
@@ -360,6 +375,298 @@ def inst_expectation(*args, **kwargs):
+            ]
+            if default_profiler_config:
+                override_profiler_config = copy.deepcopy(profiler_config)
+            else:
+                override_profiler_config = None
+
+            """
+            If default Rule-Based Profiler configuration exists, use it as base with custom Rule-Based Profiler
+            configuration as override; otherwise, use custom Rule-Based Profiler configuration with no override.
+            """
+            profiler_config = default_profiler_config or profiler_config
+
+            # Roundtrip through schema validation to remove any illegal fields add/or restore any missing fields.
+            validated_profiler_config: dict = ruleBasedProfilerConfigSchema.load(
+                profiler_config
+            )
+            profiler_config = ruleBasedProfilerConfigSchema.dump(
+                validated_profiler_config
+            )
+            profiler_config = RuleBasedProfilerConfig(**profiler_config)
+
+            profiler: BaseRuleBasedProfiler = self._build_rule_based_profiler(
+                expectation_type=expectation_type,
+                expectation_kwargs=expectation_kwargs,
+                success_keys=success_keys,
+                profiler_config=profiler_config,
+                override_profiler_config=override_profiler_config,
+            )
+
+            expectation_configurations: List[
+                ExpectationConfiguration
+            ] = profiler.run().expectations
+
+            configuration = expectation_configurations[0]
+
+            # Reconcile explicitly provided "ExpectationConfiguration" success_kwargs as overrides to generated values.
+            expectation_kwargs = configuration.kwargs
+            expectation_kwargs.update(expectation_kwargs_overrides)
+
+            if meta is None:
+                meta = {}
+
+            meta["profiler_config"] = profiler.to_json_dict()
+
+            configuration = ExpectationConfiguration(
+                expectation_type=expectation_type,
+                kwargs=expectation_kwargs,
+                meta=meta,
+            )
+        else:
+            configuration = ExpectationConfiguration(
+                expectation_type=expectation_type,
+                kwargs=expectation_kwargs,
+                meta=meta,
+            )
+
+        return configuration
+
+    def _build_rule_based_profiler(
+        self,
+        expectation_type: str,
+        expectation_kwargs: dict,
+        success_keys: Tuple[str],
+        profiler_config: RuleBasedProfilerConfig,
+        override_profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
+                    str,
+                    Union[
+                        str,
+                        float,
+                        Optional[Dict[str, Any]],
+                        Optional[Dict[str, Dict[str, Any]]],
+                    ],
+                ],
+            ]
+        ] = None,
```

Comment:
Actually, given that this is so prevalent, I'd love to discuss a way to possibly circumvent this before we merge.

Reply:
This can be a follow-on improvement.  Perhaps define the types -- but this is just our profiler/rules configuration structure (nothing new, I am just "explicitly showing what is in there so that in the future we can refactor).

Reply:
Yup don't block merge on this!